lib/vrl/parser/src/lex.rs

```diff
@@ -260,6 +260,7 @@ pub enum Token<S> {
     Escape,
 
     Equals,
+    MergeEquals,
     Bang,
     Pipe,
```

Comment:
The `Pipe` and merge `|` are the same symbol. This needs tidying up. Both path coalescing and merging seem to parse correctly, but there is the potential for conflicts and confusion..